src/components/RoomHeader/index.js

```diff
@@ -2,23 +2,44 @@ import React from 'react'
 import style from './index.module.css'
 
 export const RoomHeader = ({
-  state: { room, user, sidebarOpen, userListOpen },
-  actions: { setSidebar, setUserList },
+  state: { room, user, sidebarOpen, userListOpen, inputIsOpen, roomTitle },
+  actions: { setSidebar, setUserList, setTitleEditable, updateRoom, changeInputTitle },
 }) => (
-  <header className={style.component}>
-    <button onClick={e => setSidebar(!sidebarOpen)}>
-      <svg>
-        <use xlinkHref="index.svg#menu" />
-      </svg>
-    </button>
-    <h1>{room.name && room.name.replace(user.id, '')}</h1>
-    {room.users && (
-      <div onClick={e => setUserList(!userListOpen)}>
-        <span>{room.users.length}</span>
+    <header className={style.component}>
+      <button onClick={e => setSidebar(!sidebarOpen)}>
         <svg>
-          <use xlinkHref="index.svg#members" />
+          <use xlinkHref="index.svg#menu" />
         </svg>
-      </div>
-    )}
-  </header>
-)
+      </button>
+      {
+        (room.name && !inputIsOpen) && <h1 onClick={e => setTitleEditable(!inputIsOpen, room.name)}>
+          {room.name.replace(user.id, '')}
+        </h1>
+      }
+
+      {
+        inputIsOpen &&
+        <h1>{[
+          <form >
```

Comment:
Good practice when using forms is to add an `onSubmit` prop that, at least does `e.preventDefault` otherwise if someone hits return inside the form then the page gets reloaded.

Reply:
What exactly are you implying , I am missing your point here, do you want me to remove the svg and add a button & I haven't used onSubmit prop, I did use it in redux forms but that was different scenario. Can you elaborate a bit more ?

Reply:
Here is a form that handles both the user hitting enter and them clicking the submit button. This way the input it not _controlled_ but left to do its thing until the user submits the form. Which requires less wiring (no `changeInputTitle ` action or `roomTitle` in state).

```
<form onSubmit={e => {
  e.preventDefault()
  updateRoom(room.id, e.target.elements[0].value)
}}>
  <input />
  <button type='submit'><svg /></button>
</form>

```

Does that make sense? What do you think?